### DISCUSSION TITLE
Can you dynamically update the text in a "Reset Password" email?

### DISCUSSION THREAD

**Discussion by yingw787** [supabase/supabase#17599] [Questions] [answered]
Context: My cofounder and I are working on a loan processing startup. We're working with mortgage brokers to replace their current process of collecting documents for home mortgages. My cofounder is requesting that for the reset password flow, whether the title and text of the reset password email can come from the mortgage broker (which dynamically changes based on fields in the database), instead of a hardcoded field.

I told my cofounder that I'm not sure whether the reset-password flow is customizable to use this information, guessing reason being that JavaScript in a reset password workflow might compromise the security of the reset password workflow. I noticed in the Supabase dashboard that the reset password flow contains things like tokenHash, which I'm guessing is generated by the auth server, and I'm guessing Supabase doesn't want to have user-based, unverified JavaScript running on the auth server where the email template is generated. Practically, I don't think .resetPasswordForEmail gives an email template option in the JavaScript SDK.

One potential solution I'm thinking about is whether I can listen for the reset password event on .onAuthStateChange, and then send out a custom email to do that. But then I'd have to block the original reset password email from going out (which I don't know how to do), and figure out how to get the PKCE token from the auth server (which I don't know how to do). I guess I could go in the Supabase source code and figure out how the reset password flow works in Supabase studio (authentication tab) and then reproduce that...but that doesn't seem quite right.

How would the Supabase community handle this?

**Comment by J0** [comment, marked as answer]
Hey @yingw787,

Thanks for the query - there's a `{{ .Data }}` template tag to allow you to customize the email template based on user metadata and a [Reset Password email template](https://supabase.com/dashboard/project/_/auth/templates) not sure if this at all advances you towards your use case but do give it a look and let us know.

Thanks!
Joel
